src/wordy/ast/LoopNode.java

```diff
@@ -40,4 +44,21 @@ public int hashCode() {
     public String toString() {
         return "LoopNode{body=" + body + '}';
     }
+
+    @Override
+    protected void doRun(EvaluationContext context) {
+        while(true) {
+            try {
+                body.run(context);
+            } catch(LoopExited e) {
+                break;
```

Comment:
I see that you don't use `break`. So I guess this line of mine can be removed

Reply:
I don't remember we learned error handling in 127/128. It's a good time for me to examine error handling in Java or in programming languages in general.